src/services/http-req/dogBreedRequest.test.ts

```diff
@@ -0,0 +1,71 @@
+import fetch from 'node-fetch'
+import { getDogBreeds } from './dogBreedRequest'
+import { input1 } from '../../lambdas/mock-events/mock-values'
+
+const mockedFetch: jest.Mock = fetch as any
+
+jest.mock('node-fetch')
+
+describe('fetching api data for dog breeds', () => {
+  const mockPayload = input1
+  beforeEach(() => {
+    mockedFetch.mockReturnValueOnce({
+      json: () => {
+        return { status: 'success', message: mockPayload }
+      },
+    })
+  })
+
+  it('should return the correct payload from a sucessful fetch request', async () => {
+    const response = await getDogBreeds()
+    expect(response.status).toBe('success')
+    expect(response.message).toMatchObject(input1)
+  })
+})
+
+describe('handling errors from the external call', () => {
+  beforeEach(() => {
+    mockedFetch.mockReturnValueOnce({
+      json: () => {
+        throw new Error('I broke somewhere')
+      },
+    })
+  })
+
+  it('should bubble up an error from the external API call', async () => {
+    let response
+    try {
+      response = await getDogBreeds()
+    } catch (err) {
+      response = err
+    }
+    expect(response instanceof Error).toBe(true)
+    expect(response.message).toEqual('I broke somewhere')
+  })
+})
+
+describe('it handles the external API timing out', () => {
+  beforeEach(() => {
+    mockedFetch.mockReturnValueOnce({
+      json: () => {
+        return new Promise((resolve) => {
+          setTimeout(() => {
+            resolve({ this: 'should not matter' })
+          }, 3000)
```

Comment:
I'm not sure this `setTimeout` has any value here, we are trying to mock a return value and not an implementation. Instead we might get more value out of returning the error we're expecting from fetch and checking that our code works off that. This way we can also manipulate the error to match something that an abort controller would throw 🙂

Reply:
I see what you're doing here though, the timeout in the code is 1500 and you're saying this should wait 3000, that's pretty clever but there are cleaner ways to accomplish this 👍

Reply:
I think this comment was my aha moment about what I was doing wrong, thank you. Since the API call is being mocked we won't see an actual AbortError from the abort controller since the listener attached to the fetch isn't there. I believe this means the test doesn't need to worry about a delay, it just needs to make sure any errors that occur in this function are passed on correctly.